crates/storage/provider/src/providers/rocksdb_stub.rs

```diff
@@ -0,0 +1,128 @@
+//! Stub implementation of `RocksDB` provider.
+//!
+//! This module provides placeholder types that allow the code to compile when `RocksDB` is not
+//! available (either on non-Unix platforms or when the `rocksdb` feature is not enabled).
+//! Operations will produce errors if actually attempted.
+
+use reth_db_api::table::{Decompress, Table};
+use reth_storage_errors::{
+    db::LogLevel,
+    provider::{ProviderError::UnsupportedProvider, ProviderResult},
+};
+use std::path::Path;
+
+/// A stub `RocksDB` provider.
+///
+/// This type exists to allow code to compile when `RocksDB` is not available (either on non-Unix
+/// platforms or when the `rocksdb` feature is not enabled). When using this stub, the
+/// `transaction_hash_numbers_in_rocksdb` flag should be set to `false` to ensure all operations
+/// route to MDBX instead.
+#[derive(Debug, Clone)]
+pub struct RocksDBProvider;
+
+impl RocksDBProvider {
+    /// Creates a new stub `RocksDB` provider.
+    ///
+    /// On non-Unix platforms, this returns an error indicating `RocksDB` is not supported.
+    pub fn new<P: AsRef<Path>>(_path: P) -> ProviderResult<Self> {
+        Ok(Self)
+    }
+
+    /// Creates a new stub `RocksDB` provider builder.
+    pub fn builder<P: AsRef<Path>>(path: P) -> RocksDBBuilder {
+        RocksDBBuilder::new(path)
+    }
+
+    /// Get a value from `RocksDB` (stub implementation).
+    pub fn get<T>(&self, _key: T::Key) -> ProviderResult<Option<T::Value>>
+    where
+        T: Table,
+        T::Value: Decompress,
+    {
+        Err(UnsupportedProvider)
+    }
+
+    /// Put a value into `RocksDB` (stub implementation).
+    pub fn put<T>(&self, _key: T::Key, _value: T::Value) -> ProviderResult<()>
```

Comment:
changed to match the signature of the real impl
```suggestion
    pub fn put<T>(&self, _key: T::Key, _value: &T::Value) -> ProviderResult<()>
```